src/controller/CHIPDeviceController.cpp

```diff
@@ -1724,13 +1726,49 @@ void DeviceCommissioner::OnDone()
     CommissioningStageComplete(return_err, report);
 }
 
+// TODO if we want to branch out the Commissioning Error to a specific chip error this will be
+// moved to the top and likely is better as a map lookup (which I would look for suggestions).
+CHIP_ERROR ConvertCommissioningErrorToChipError(GeneralCommissioning::CommissioningError commissioningError)
+{
+    switch (commissioningError)
+    {
+    case GeneralCommissioning::CommissioningError::kOk:
+        return CHIP_NO_ERROR;
+        break;
+    case GeneralCommissioning::CommissioningError::kValueOutsideRange:
+        return CHIP_ERROR_INVALID_ARGUMENT; //  TODO figure out more appropriate error.
+        break;
+    case GeneralCommissioning::CommissioningError::kInvalidAuthentication:
+        return CHIP_ERROR_INVALID_ARGUMENT; //  TODO figure out more appropriate error.
+        break;
+    case GeneralCommissioning::CommissioningError::kNoFailSafe:
+        return CHIP_ERROR_INVALID_ARGUMENT; //  TODO figure out more appropriate error.
+        break;
+    case GeneralCommissioning::CommissioningError::kBusyWithOtherAdmin:
+        return CHIP_ERROR_INVALID_ARGUMENT; //  TODO figure out more appropriate error.
+        break;
+    default:
+        break;
+    }
+    return CHIP_ERROR_INVALID_ARGUMENT; //  TODO figure out more appropriate error.
+}
+
 void DeviceCommissioner::OnArmFailSafe(void * context,
                                        const GeneralCommissioning::Commands::ArmFailSafeResponse::DecodableType & data)
 {
-    // TODO: Use errorCode
-    ChipLogProgress(Controller, "Received ArmFailSafe response");
+    // TODO: Figure out if ther is a better way other than static_cast.
+    ChipLogProgress(Controller, "Received ArmFailSafe response errorCode=%u", static_cast<uint8_t>(data.errorCode));
+
+    CommissioningDelegate::CommissioningReport report;
+    if (data.errorCode != GeneralCommissioning::CommissioningError::kOk)
+    {
+        report.Set<CommissionErrorInfo>(data.errorCode);
+    }
+
+    // TODO: do we actually want to look this up or is it okay to just use CHIP_ERROR_INTERNAL instead of this mapping
```

Comment:
Up to you. If you're sending a report with the actual cluster error, the mapping is a bit redundant.

Reply:
Yeah I think that it is pretty redundant, removing it.